native/java/org/jpype/JPypeSignal.java

```diff
@@ -0,0 +1,47 @@
+package org.jpype;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+/**
+ * Java wants to make this action nearly impossible.
+ *
+ * Thus the have warnings against it that cannot be disabled. So we will skin
+ * this cat another way.
+ */
+public class JPypeSignal
+{
+
+  static Thread main;
+
+  static void installHandlers()
+  {
+    try
+    {
+      Class Signal = Class.forName("sun.misc.Signal");
+      Class SignalHandler = Class.forName("sun.misc.SignalHandler");
```

Comment:
Is this signal handler also abundant in other JVMs?

Reply:
Why isnt the usual shutdown hook mechanism sufficient for SIGINT?

Reply:
That is a tough one to answer.  I haven't got much vision into others.  But I suspect that ``sun.misc.Signal`` and ``sun.misc.Unsafe`` are both prevalent in many versions.   I guess I should misname this misc2 and make sure we will still run even if the hooks are missing.

Reply:
The problem here is that people don't want to shutdown here.   If someone is running a procedure under interactive Python and they hit ^C they are expecting a break out of the loop or whatever is running and a return to the interactive console.  But Java is catching the ^C because it has taken over all the signal handlers.   So it takes the ^C as a signal to kill everything (which would be fine if it was in charge), and starts a shutdown sequence.  Python doesn't know anything about it and all of sudden the stack frames in Java start collapsing.  Thus a crash is likely to happen.   The correct action is for Java to take the ^C as a hint to and let Python decide if it wants to shutdown (after all it is in charge here).   If Python wants to shutdown then shutdownJVM happens and everything is normal.   

I suppose we can add a keyword argument to startJVM ``interactive=False`` which will allow ^C to proceed with the default behavior.   But 90% of the time Java shutting down without telling Python about it first is likely to just go straight to a crash.

Reply:
Would it also be possible to unregister all signal handlers, which the JVM registered during startup, and let only Python handle signals? This way we would not need to take care about different JVM implementations and have full control of the process.
Probably I'm currently unaware of all possible consequences, but it might be worth a consideration. What do you think?

Reply:
I am not sure that really works.   The JVM is pretty darn insistent about its signal handlers.  Several of them a vital for operation.  We could repoint just this one back to C++ by capturing and restoring this one handler. 

But to do that we would have to understand how Python and Java are directing signals (when mixed with threads).   My understanding is that Python is redirecting all signals to the main thread and then calling the signal routines.   Java on the other hand spins a thread up for each single.

It is also not clear how the signals were layered in the first place.  When we hit control C in Python it is clear that both Python and Java got the signal.  I assume to came to Java who chained it to Python, but I don't have technical details.

Do you want me to try to manually control the signals for this PR?   If so I may have to deal with OS specific behaviors as windows and unix have different concepts of signal handling.

Reply:
Thanks for pointing out, that my idea is infeasible to implement. I was just thinking about per process signals, and forgot about the multithreaded path.